src/Symfony/Component/Serializer/Context/ContextInterface.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Symfony\Component\Serializer\Context;
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+interface ContextInterface
```

Comment:
Do we need this interface? Cannot we use the builder pattern as done in HttpClient (the `HttpOptions` helper)  instead? This new class should contain a method for every available options for autocompletion, as in HttpClient.